### DISCUSSION TITLE
How can we mock database calls from Eloquent models for unit testing in laravel?

### DISCUSSION THREAD

**Discussion by drewg2009** [laravel/framework#45560] [Q&A]
Recently our team has been using https://github.com/imanghafoori1/eloquent-mockery to mock database calls to improve test runtime as well as to isolate testing to business logic. 

However this library is very small, with limited support, and just underwent a major refactor that broke some of our tests. 

Looking the laravel docs for database testing https://laravel.com/docs/master/database-testing, it looks like we can create test records but we cannot mock the eloquent calls.

Without using this approach yet it seems like this would run a lot slower in our pipeline but it would be a good solution moving forward. 

Has anyone been able to mock database calls for the query builder or Eloquent ORM calls? If so is there documentation on this? 

If not is is best to just use the actual database with test records?

**Comment by henzeb** [comment]
You could run tests with  sqllite which is a similar setup as you have now.

**Reply by drewg2009** [reply]
does that actually mock the database calls? it seems like that would just be a test database so we would actually be transacting with a database in our pipeline. the goal is to mock the database calls if possible

**Reply by henzeb** [reply]
Sqllite is file based. The driver for laravel for it is complete. It may have some differences with the database you are actually using, but for most situations this is doable.
